app/collections/collaborators.jsx

```diff
@@ -0,0 +1,375 @@
+import React from 'react';
+import apiClient from 'panoptes-client/lib/api-client';
+import UserSearch from '../components/user-search';
+import checkIfCollectionOwner from '../lib/check-if-collection-owner';
+import alert from '../lib/alert';
+
+const ID_PREFIX = 'COLLECTION_COLLABORATORS_PAGE_';
+
+const POSSIBLE_ROLES = [
+  'owner',
+  'collaborator',
+  'contributor',
+  'viewer'
+];
+
+const ROLES_INFO = {
+  owner: {
+    label: 'Owner',
+    description: 'The collection creator. There can only be one.'
+  },
+  collaborator: {
+    label: 'Collaborator',
+    description: 'Collaborators have full access to add and remove subjects from the collection.'
+  },
+  contributor: {
+    label: 'Contributor',
+    description: 'Contributors can only add subjects to the collection.'
+  },
+  viewer: {
+    label: 'Viewer',
+    description: 'Viewers can see this collection even if it\'s private.'
+  }
+};
+
+export class RoleCreator extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      creating: null,
+      error: null
+    };
+
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+
+    const checkboxes = this.node.querySelectorAll('[name="role"]');
+    const users = this.userSearch.value().map(option => parseInt(option.value, 10));
+    const checkedBoxes = Object.keys(checkboxes).filter(key => checkboxes[key].checked);
+    const roles = checkedBoxes.map(value => checkboxes[value].value);
+
+    this.setState({
+      creating: true,
+      error: null
+    });
+
+    const roleSets = users.map(id =>
+      apiClient.type('collection_roles').create({
+        roles,
+        links: {
+          collection: this.props.collection.id,
+          user: id
+        }
+      })
+    );
+
+    Promise.all((roleSets.map((roleSet) => {
+      return roleSet.save().then(role => role);
```

Comment:
Ahh. Good catch. I don't think it's necessary. I was initially having issues with this `Promise.all` returning correctly, but I can safely remove the `then()`.